pkg/vault/contracts/AggregatorsRouter.sol

```diff
@@ -0,0 +1,170 @@
+    ) internal returns (uint256 amountCalculated, uint256 amountIn, uint256 amountOut) {
+        // The deadline is timestamp-based: it should not be relied upon for sub-minute accuracy.
+        // solhint-disable-next-line not-rely-on-time
+        if (block.timestamp > params.deadline) {
+            revert SwapDeadline();
+        }
+
+        (amountCalculated, amountIn, amountOut) = _vault.swap(
+            VaultSwapParams({
+                kind: params.kind,
+                pool: params.pool,
+                tokenIn: params.tokenIn,
+                tokenOut: params.tokenOut,
+                amountGivenRaw: params.amountGiven,
+                limitRaw: params.limit,
+                userData: params.userData
+            })
+        );
+    }
+
+    /*******************************************************************************
+                                      Queries
+    *******************************************************************************/
+
+    /// @inheritdoc IAggregatorsRouterRouter
+    function querySwapSingleTokenExactIn(
+        address pool,
+        IERC20 tokenIn,
+        IERC20 tokenOut,
+        uint256 exactAmountIn,
+        address sender,
+        bytes memory userData
+    ) external saveSender(sender) returns (uint256 amountCalculated) {
+        return
+            abi.decode(
+                _vault.quote(
+                    abi.encodeCall(
+                        AggregatorsRouter.querySwapHook,
+                        IRouter.SwapSingleTokenHookParams({
+                            sender: msg.sender,
+                            kind: SwapKind.EXACT_IN,
+                            pool: pool,
+                            tokenIn: tokenIn,
+                            tokenOut: tokenOut,
+                            amountGiven: exactAmountIn,
+                            limit: 0,
+                            deadline: _MAX_AMOUNT,
+                            wethIsEth: false,
+                            userData: userData
+                        })
+                    )
+                ),
+                (uint256)
+            );
+    }
+
+    /**
+     * @notice Hook for swap queries.
+     * @dev Can only be called by the Vault. Also handles native ETH.
+     * @param params Swap parameters (see IRouter for struct definition)
+     * @return amountCalculated Token amount calculated by the pool math (e.g., amountOut for a exact in swap)
+     */
+    function querySwapHook(
+        IRouter.SwapSingleTokenHookParams calldata params
+    ) external nonReentrant onlyVault returns (uint256) {
+        (uint256 amountCalculated, , ) = _swapHook(params);
+
+        return amountCalculated;
+    }
+
+    /***************************************************************************
+                                       Overrides
+    ***************************************************************************/
+    function permitBatchAndCall(
```

Comment:
```suggestion
    ***************************************************************************/

    function permitBatchAndCall(
```
Are we not supporting because it's payable? What about `multicall`? Should we also override and reject that? Or only reject them if msg.value > 0?

If we are reverting, I think `OperationNotSupported` can just be local to the AggregatorRouter (not in the interface), as it's not an error a proper user should get. And I don't think it needs an argument.